packages/tools/devtools/devtools-core/src/data-visualization/VisualSharedTreeTypes.ts

```diff
@@ -27,13 +27,21 @@ export interface SharedTreeSchemaNode {
 	/**
 	 * Name of the SharedTree schema.
 	 */
-	schemaName: string;
+	readonly schemaName: string;
 
 	/**
 	 * Types allowed (e.g., string, number, boolean, handle & etc.) inside the node.
 	 * - InternalNode: `Record<string, string>`.
 	 */
-	allowedTypes?: string | Record<string, string>;
+	readonly allowedTypes?: string | Record<string, string>;
```

Comment:
When will this be undefined? What does it mean?

Reply:
Removed optional type for `allowedTypes`.